addons/Trusts/Trusts.lua

```diff
@@ -0,0 +1,338 @@
+    else
+      calls = limit - #party_ind - others
+    end
+
+    repeat
+      local index = trusts[math.random(1,#trusts)]
+      if not table.find(checked,index.name) then
+        table.insert(checked,index.name)
+        if windower.ffxi.get_spells()[index.id] and windower.ffxi.get_spell_recasts()[index.id] == 0 then
+          table.insert(queue,index)
+        end
+      end
+    until #queue >= calls or #checked >= 103 --# of unique names w/o Cornelia
+
+  elseif settings.sets[set] then
+    retr = {unpack(party_ind)}
+    for i=1,limit do
+      if settings.sets[set][tostring(i)] then
+        local entity = trusts:with(lang,settings.sets[set][tostring(i)])
+        if not party[entity.name]
+        or party[entity.name] ~= entity.models then
+          if windower.ffxi.get_spell_recasts()[entity.id] == 0 then
+            if windower.ffxi.get_spells()[entity.id] then
+              table.insert(queue,entity)
+            else
+              table.remove(retr,table.find(retr,party_ind[i]))
+              error('You aren\'t trusted by '..entity.english..'.')
+            end
+          else
+            table.remove(retr,table.find(retr,party_ind[i]))
+            local recast = math.floor(windower.ffxi.get_spell_recasts()[entity.id] / 6) / 10
+            log(entity.english..' needs '..recast..' secs break.')
+          end
+        else
+          table.remove(retr,table.find(retr,entity.name))
+          if settings.auto then
+            log(entity.english..' already exists.')
+          end
+        end
+      end
+    end
+    for index,name in pairs(retr) do
+      if #retr == #party_ind then
+        windower.send_command('input /retr all')
+        coroutine.sleep(settings.wait.retrall)
+        break;
+      else
+        windower.send_command('input /retr '..name)
+        coroutine.sleep(settings.wait.retr)
+      end
+    end
+  else
+      error('Unknown set name '..(set or ''))
+  end
+  --if /retr then wait at least 3secs.
+  local delay = (limit - #party_ind) == 0 and math.max(0,settings.wait.retrall + time - os.clock()) or 0
+  coroutine.schedule(call_trust,delay)
+end
+
+windower.register_event('action', function(act)
+  if settings.auto and act.actor_id == windower.ffxi.get_player().id and queue and #queue > 0 then
+    if act.category == 4 and act.param == table.remove(queue,1).id then
+      coroutine.schedule(call_trust,settings.wait.aftercast)
+    elseif act.category == 8 and act.param == 28787 and act.targets[1].actions[1].param == queue[1].id then
+      coroutine.schedule(call_trust,settings.wait.aftercast)
+    end
+  end
+end)
+
+function check_learned()
+  local learned = {}
+  for i,value in ipairs(trusts) do
+    if windower.ffxi.get_spells()[value.id] == false and not value.english:endswith('(UC)') then
+      table.insert(learned,value.id)
+      log(check_lang(value))
+    end
+  end
+  log('You haven\'t trusted yet from '..#learned..' trusts.')
+end
+
+trusts = T{
```

Comment:
You could probably use the resources, but I assume we don't have `models` in there? If not, that's something that would be worth adding to our `fixes.xml` in the ResourceExtractor repository... consider adding that if you want.

Reply:
done. https://github.com/Windower/ResourceExtractor/pull/53